synthesizer/src/vm/finalize.rs

```diff
@@ -1875,115 +1939,1009 @@ finalize compute:
+        vm.add_next_block(&genesis).unwrap();
+
+        // Check that the state of the `credits.aleo` program is correct.
+        let program_id = ProgramID::from_str("credits.aleo").unwrap();
+        let committee_mapping_name = Identifier::from_str("committee").unwrap();
+        let account_mapping_name = Identifier::from_str("account").unwrap();
+        let bonded_mapping_name = Identifier::from_str("bonded").unwrap();
+        let metadata_mapping_name = Identifier::from_str("metadata").unwrap();
+        let unbonding_mapping_name = Identifier::from_str("unbonding").unwrap();
+
+        // Get the committee mapping.
+        let actual_committee = vm.finalize_store().get_mapping_confirmed(program_id, committee_mapping_name).unwrap();
+        let expected_committee = committee
+            .members()
+            .iter()
+            .map(|(address, (amount, is_open))| {
+                (
+                    Plaintext::from_str(&address.to_string()).unwrap(),
+                    Value::from_str(&format!("{{ microcredits: {amount}u64, is_open: {is_open} }}")).unwrap(),
+                )
+            })
+            .collect_vec();
+        assert_eq!(actual_committee.len(), expected_committee.len());
+        for entry in actual_committee.iter() {
+            assert!(expected_committee.contains(entry));
+        }
+
+        // Get the account mapping.
+        let actual_account = vm.finalize_store().get_mapping_confirmed(program_id, account_mapping_name).unwrap();
+        let expected_account = public_balances
+            .iter()
+            .map(|(address, amount)| {
+                (Plaintext::from_str(&address.to_string()).unwrap(), Value::from_str(&format!("{amount}u64")).unwrap())
+            })
+            .collect_vec();
+        assert_eq!(actual_account.len(), expected_account.len());
+        // Check that all entries except for the one for the first validator are the same.
+        for entry in actual_account.iter() {
+            if entry.0
+                != Plaintext::from_str(&Address::try_from(validators.keys().next().unwrap()).unwrap().to_string())
+                    .unwrap()
+            {
+                assert!(expected_account.contains(entry));
+            }
+        }
+
+        // Get the bonded mapping.
+        let actual_bonded = vm.finalize_store().get_mapping_confirmed(program_id, bonded_mapping_name).unwrap();
+        let expected_bonded = bonded_balances
+            .iter()
+            .map(|(address, (validator, amount))| {
+                (
+                    Plaintext::from_str(&address.to_string()).unwrap(),
+                    Value::from_str(&format!("{{ validator: {validator}, microcredits: {amount}u64 }}")).unwrap(),
+                )
+            })
+            .collect_vec();
+        assert_eq!(actual_bonded.len(), expected_bonded.len());
+        for entry in actual_bonded.iter() {
+            assert!(expected_bonded.contains(entry));
```

Comment:
Why don't we check the amounts in the bonded and account mappings?

Reply:
NVM, i see now that the `actual_bonded` and `expected_bonded` variables are Vectors, so the contains call will check that the balance is correct.